### ISSUE TITLE
I want to separate handwritten text from printed text. Which config to set for Text - Image Data !? 

### ISSUE THREAD

**Issue by harshgrovr** [matterport/Mask_RCNN#1703]
**Does Nucleus config will work?** Which config will do better? 

**My Data:** 

![0002_grimm_donner_1855_0002](https://user-images.githubusercontent.com/8547940/63552833-0f2e2180-c539-11e9-9752-44ea6e73ef36.jpg)

**Currently My Config is:**

    GPU_COUNT = 1
    IMAGES_PER_GPU = 1

    # Number of classes (including background)
    NUM_CLASSES = 1 + 1  # background + 3 shapes

    # Use small images for faster training. Set the limits of the small side
    # the large side, and that determines the image shape.
    
    IMAGE_MIN_DIM = 1024
    IMAGE_MAX_DIM = 1024
    
    POST_NMS_ROIS_TRAINING = 1000
    POST_NMS_ROIS_INFERENCE = 2000
    
     # Non-max suppression threshold to filter RPN proposals.
    # You can increase this during training to generate more 
    #propsals.
    RPN_NMS_THRESHOLD = 0.9

    IMAGE_RESIZE_MODE = "crop"
    IMAGE_MIN_SCALE = 2.0
    
    # How many anchors per image to use for RPN training
    RPN_TRAIN_ANCHORS_PER_IMAGE = 320
    MEAN_PIXEL = np.array([43.53, 39.56, 48.22])
    USE_MINI_MASK = True
    MINI_MASK_SHAPE = (56, 56)  # (height, width) of the mini-mask
    TRAIN_ROIS_PER_IMAGE = 128
    MAX_GT_INSTANCES = 600
    DETECTION_MAX_INSTANCES = 400

    # Use smaller anchors because our image and objects are small
    RPN_ANCHOR_SCALES = (8, 16, 32, 64, 128)  # anchor side in pixels

    # Reduce training ROIs per image because the images are small and have
    # few objects. Aim to allow ROI sampling to pick 33% positive ROIs.

    # Use a small epoch since the data is simple
    STEPS_PER_EPOCH = 300

    # use small validation steps since the epoch is small
    VALIDATION_STEPS = 35

I used coco weights and trained the network for 300 epocs. I am getting mean AP as 0.65 

**Please help from your valuable time.**

**Comment by AGG-Github** [comment]
dude, which tool do you use to mask the picture?

**Comment by harshgrovr** [comment]
GT was already annotated, it was already given to me by someone.

**Comment by hpanwar08** [comment]
@harshgrovr did you get it to work? Can you show this image with all the annotations (total number of classes)? I my opinion you should have total 3 classes, 1for background, 1 for handwritten text and 1 for printed text.  IMAGE_MIN_DIM, IMAGE_MAX_DIM  can be determined from the sizes of your images in your dataset.

**Comment by prashant-tft** [comment]
Hi @harshgrovr ,

Have you found any way to solve it. Please reply as it is very urgent for me.

**Comment by harshgrovr** [comment]
No, I had to use ensembling method, which involves training mask_rcnn with various configuration and then combining output mask of all configuration with original image. And then Training Maskrcnn again with that ensembled output.

You can use any other segmentation network for the last part.

**Comment by adesgautam** [comment]
Without ensembling, wouldn't the mask_rcnn work ? Is it giving FP too ?

**Comment by harshgrovr** [comment]
The size of dataset was less hence we have to do all this. Not sure about other datasets though.

**Comment by adesgautam** [comment]
Okay. I tried on around 200 images, the results were okay for handwriting detection. And there were FPs too. Maybe with more data it would improve.
